wallet/src/lib.rs

```diff
@@ -46,6 +48,23 @@ pub struct Admin {
     pub weight: u32,
 }
 
+#[contracterror]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum Error {
+    // Initialization errors
+    InvalidAdminWeight = 1,
+    AdminWeightsBelowThreshold = 2,
+    InvalidAdminCount = 3,
+    InvalidThreshold = 4,
+    AlreadyInitialized = 5,
+    // Payment errors
+    StoredPaymentMismatch = 6,
+    DuplicateSigner = 7,
+    UnauthorizedSigner = 8,
+    PaymentAlreadyExecuted = 9,
+}
```

Comment:
I don't have a strong opinion on this. I imagine splitting initialization-exclusive errors from everything else could be useful and wouldn't hurt. For contracts with more than just 1 non-init fn I probably wouldn't go further with splitting the error groups. We also should consider how hard would it be to consume and decode the contract error when/if we start returning it for the failed txs.

Reply:
>We also should consider how hard would it be to consume and decode the contract error when/if we start returning it for the failed txs.

That's a good point. Also, the 'u32' naemspace of contract errors is shared across all types, so even if they were two separaet enums, it would be a bad idea to reuse integer values between them.